optimum/onnxruntime/trainer.py

```diff
@@ -260,13 +309,63 @@ def __init__(
             optimizers=optimizers,
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
-
+        self._training_model = ModuleWithLoss(model, args)
+        self.model = model
+        self._inferencing_model = model
         self.feature = feature
         self.onnx_model_path = onnx_model_path
         self.exported_with_loss = False
         if self.args.local_rank:
             torch.cuda.set_device(self.args.local_rank)
 
+    def _set_signature_columns_if_needed(self):
+        if self._signature_columns is None:
```

Comment:
add a note saying we assume training_model and inference_model assume the same forward signature_column. So far, self._signature_columns only store the first-time parsed signature, whatever it is from training path or inference path.